syntax = "proto3";

option java_multiple_files = true;
option java_package = "com.narc.arclient.network";
option java_outer_classname = "RemoteRecognizeServiceProto";

package network;

service RemoteRecognizeService {
  rpc recognize (RecognizeRequest) returns (RecognizeResponse) {}
  rpc logReport(LogReportRequest) returns (EmptyResponse) {}
  rpc systemState(SystemStateRequest) returns (EmptyResponse) {}
}

message LogReportRequest {
  string deviceSerialNumber = 1;
  string taskId = 2;
  string unloadEnd = 3;
  string startTime = 4;
  string endTime = 5;
  string posExist = 6;
  int64 copyTime = 7;
  int64 preprocessTime = 8;
  int64 recognizeTime = 9;
  int64 renderTime = 10;
  int64 transfer2RemoteTime = 11;
  int64 computeRemoteTime = 12;
  int64 transfer2LocalTime = 13;
}

message SystemStateRequest {
  string deviceSerialNumber = 1;
  int64 timestamp = 2;
  string cpuUsage = 3;
  string memUsage = 4;
  string batteryLevel = 5;
}

message EmptyResponse {
  string success = 1;
}

message RecognizeRequest {
  bytes bitmap_data = 1;
}

message RecognizeResponse {
  string handedness = 1;
  string gesture = 2;
  float x1 = 3;
  float y1 = 4;
  float x2 = 5;
  float y2 = 6;
  int64 recieve_time = 7;
  int64 sendback_time = 8;
}
